deploy/kubelessDeploy.js

```diff
@@ -48,87 +50,118 @@ class KubelessDeploy {
+          path.join(this.serverless.config.servicePath || '.', relativePath),
+          (err, d) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(d.toString());
+            }
+          });
       });
     }
     return resultPromise;
   }
 
-  deployFunction() {
-    const thirdPartyResources = new Api.ThirdPartyResources(
+  getThirdPartyResources() {
+    return new Api.ThirdPartyResources(
       Object.assign(helpers.getMinikubeCredentials(), {
         url: process.env.KUBE_API_URL,
         group: 'k8s.io',
       })
     );
+  }
+
+  deployFunction() {
+    const thirdPartyResources = this.getThirdPartyResources();
     thirdPartyResources.addResource('functions');
+
+    let files = {
+      handler: null,
+      deps: null,
+    };
     const errors = [];
     let counter = 0;
     return new BbPromise((resolve, reject) => {
       _.each(this.serverless.service.functions, (description, name) => {
-        this.getFunctionContent(
-          `${description.handler.toString().split('.')[0]}.py`
-        ).then(functionContent => {
-          const funcs = {
-            apiVersion: 'k8s.io/v1',
-            kind: 'Function',
-            metadata: {
-              name,
-              namespace: 'default',
-            },
-            spec: {
-              deps: '',
-              function: functionContent,
-              handler: description.handler,
-              runtime: this.serverless.service.provider.runtime,
-              topic: '',
-              type: 'HTTP',
-            },
+        if (this.serverless.service.provider.runtime.match(/python/)) {
+          files = {
+            handler: `${description.handler.toString().split('.')[0]}.py`,
+            deps: 'requirements.txt',
           };
-          // Create function
-          thirdPartyResources.ns.functions.post({ body: funcs }, (err) => {
-            if (err) {
-              if (err.code === 409) {
-                this.serverless.cli.log(
-                  `The function ${name} is already deployed. ` +
-                  'Remove it if you want to deploy it again.'
-                );
-              } else {
-                errors.push(
-                  `Unable to deploy the function ${name}. Received:\n` +
-                  `  Code: ${err.code}\n` +
-                  `  Message: ${err.message}`
-                );
-              }
-            } else {
-              this.serverless.cli.log(
-                `Function ${name} succesfully deployed`
-              );
-            }
-            counter++;
-            if (counter === _.keys(this.serverless.service.functions).length) {
-              if (_.isEmpty(errors)) {
-                resolve();
-              } else {
-                reject(
-                  `Found errors while deploying the given functions:\n${errors.join('\n')}`
-                );
-              }
-            }
+        } else {
```

Comment:
Just a comment. I don't know how difficult would be to extend this for other runtimes. For example, I don' know if kubeless support installing dependencies from a package.json file for nodejs function